test/integration/client-side-encryption/driver.test.ts

```diff
@@ -0,0 +1,297 @@
+        .then(() => encryptedClient?.close())
+        .then(() => client?.close());
+    });
+
+    const testCases = [
+      {},
+      { promoteValues: true },
+      { promoteValues: false },
+      { promoteValues: true, promoteLongs: false },
+      { promoteValues: true, promoteLongs: true },
+      { bsonRegExp: true },
+      { ignoreUndefined: true }
+    ];
+
+    for (const bsonOptions of testCases) {
+      const name = `should respect bson options ${JSON.stringify(bsonOptions)}`;
+
+      it(name, metadata, async function () {
+        const data = {
+          _id: new BSON.ObjectId(),
+          a: 12,
+          b: new BSON.Int32(12),
+          c: new BSON.Long(12),
+          d: new BSON.Double(12),
+          e: /[A-Za-z0-9]*/,
+          f: new BSON.BSONRegExp('[A-Za-z0-9]*'),
+          g: undefined
+        };
+
+        const expected = BSON.deserialize(BSON.serialize(data, bsonOptions), bsonOptions);
+
+        const coll = encryptedClient.db(dataDbName).collection(dataCollName);
+        const result = await coll.insertOne(data, bsonOptions);
+        const actual = await coll.findOne({ _id: result.insertedId }, bsonOptions);
+        const gValue = actual?.g;
+        delete actual?.g;
+
+        expect(actual).to.deep.equal(expected);
+        expect(gValue).to.equal(bsonOptions.ignoreUndefined ? data.g : null);
+      });
+    }
+  });
+
+  describe('key order aware command properties', () => {
+    let client: MongoClient;
+    let collection: Collection;
+
+    beforeEach(async function () {
+      if (!this.configuration.clientSideEncryption.enabled) {
+        return;
+      }
+
+      const encryptionOptions = {
+        monitorCommands: true,
+        autoEncryption: {
+          keyVaultNamespace,
+          kmsProviders: { local: { key: 'A'.repeat(128) } }
+        }
+      };
+      client = this.configuration.newClient({}, encryptionOptions);
+      collection = client.db(dataDbName).collection('keyOrder');
+    });
+
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    describe('find', () => {
+      it('should maintain ordered sort', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const sort: ReadonlyArray<[string, SortDirection]> = Object.freeze([
+          ['1', 1],
+          ['0', 1]
+        ]);
+        await collection.findOne({}, { sort });
+        const findEvent = events.find(event => !!event.command.find);
+        expect(findEvent).to.have.property('commandName', 'find');
+        expect(findEvent).to.have.nested.property('command.sort').deep.equal(new Map(sort));
+      });
+    });
+
+    describe('findAndModify', () => {
+      it('should maintain ordered sort', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const sort: [string, SortDirection][] = [
```

Comment:
pointing out this case since you only addressed the freeze in the `find` function test case

Reply:
fixed